cmd/kubelet/app/server.go

```diff
@@ -580,25 +582,27 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.Dependencies, stopCh <-chan
 		kubeDeps.OnHeartbeatFailure = closeAllConns
 
 		// create clients for each tenant partition
-		klog.V(6).Infof("make kubeDeps.KubeTPClients based on TenantPartitionApiservers args: %v", s.TenantPartitionApiservers)
-		if s.TenantPartitionApiservers == nil || len(s.TenantPartitionApiservers) == 0 {
-			klog.Infof("TenantPartitionApiservers not set. Default to single tenant partition and clientConfig setting")
-			s.TenantPartitionApiservers = make([]string, 1)
-			s.TenantPartitionApiservers[0] = clientConfigs.GetConfig().Host
-		}
-
-		clientConfigCopy := *restclient.CopyConfigs(clientConfigs)
-		kubeDeps.KubeTPClients = make([]clientset.Interface, len(s.TenantPartitionApiservers))
-
-		for i, tenantServer := range s.TenantPartitionApiservers {
-			for _, cfg := range clientConfigCopy.GetAllConfigs() {
-				cfg.Host = tenantServer
-				klog.V(6).Infof("clientConfigCopy.Host: %v", cfg.Host)
-			}
-
-			kubeDeps.KubeTPClients[i], err = clientset.NewForConfig(&clientConfigCopy)
+		klog.V(3).Infof("make kubeDeps.KubeTPClients based on TenantPartitionKubeConfig arg: %v", s.TenantPartitionKubeConfig)
+		if len(s.TenantPartitionKubeConfig) == 0 {
```

Comment:
if the stringVar is used for the kubeconfig parameter, better just check if it is empty "" string as default. functionality the same. directly compare string empty is clearer -- in contrast the len(s) to check array size as it was before as TenantPartitionApiServers.

Reply:
https://stackoverflow.com/questions/18594330/what-is-the-best-way-to-test-for-an-empty-string-in-go#:~:text=As%20pointed%20out%20in%20Timmmm's,identical%20code%20in%20both%20cases.&text=because%20the%20semantics%20is%20clearly%20equal.&text=Checking%20for%20length%20is%20a,that%20is%20also%20only%20whitespace.

Historically, check length is faster than compare with string. But golang has optimization to generate same code in this case. So it becomes a matter of taste.

Scheduler check "" and KCM checks length. I will keep this implementation.

Reply:
that is fine. i was not worried about this micro optimization here. it was due to the before this change, the parameter is slice so check length of slice indicates how many kubeconfigs are passed in. 
we can resolve. no sweat.